docs/core/integration-quality-scale/rules/iqs020.md

```diff
@@ -0,0 +1,97 @@
+---
+title: "IQS020 - Reauthentication flow"
+---
+
+IQS020 is as follows:
+
+Reauthentication flow
+
+## Reasoning
+
+It can happen that users change their password of a device or service and forget that their device or account is still linked to Home Assistant.
+To avoid that the user has to remove the configuration entry and re-add it, we start a reauthentication flow.
+During this flow, the user can provide the new credentials to use from now on.
+
+This is a very user-friendly way to let the user know that they need to take action and update their credentials.
+
+## Example implementation
+
+In the example below, we show an authentication flow that allows the user to reauthenticate with a new API token.
+When we receive the new token, we check if we can connect to the service to avoid the user from entering an invalid token.
+If the connection is successful, we update the configuration entry with the new token.
+
+`config_flow.py`:
+```python
+class MyConfigFlow(ConfigFlow, domain=DOMAIN):
+    """My config flow."""
+
+    host: str | None = None
+    verify_ssl: bool = True
```

Comment:
Don't include verify ssl. We should just include things that are needed to make the example clear.